manim/mobject/types/vectorized_mobject.py

```diff
@@ -2229,31 +2272,106 @@ def construct(self):
     """
 
     def __init__(
-        self, vmobject, num_dashes=15, dashed_ratio=0.5, color=WHITE, **kwargs
+        self,
+        vmobject,
+        num_dashes=15,
+        dashed_ratio=0.5,
+        dash_offset=0,
+        color=WHITE,
+        even_lengths=True,
+        **kwargs,
     ):
 
         self.dashed_ratio = dashed_ratio
         self.num_dashes = num_dashes
         super().__init__(color=color, **kwargs)
         r = self.dashed_ratio
         n = self.num_dashes
-        if num_dashes > 0:
+        if n > 0:
             # Assuming total length is 1
             dash_len = r / n
             if vmobject.is_closed():
                 void_len = (1 - r) / n
             else:
-                void_len = (1 - r) / (n - 1)
+                if n == 1:
+                    void_len = 1 - r
+                else:
+                    void_len = (1 - r) / (n - 1)
 
-            self.add(
-                *(
-                    vmobject.get_subcurve(
-                        i * (dash_len + void_len),
-                        i * (dash_len + void_len) + dash_len,
-                    )
+            period = dash_len + void_len
+            phase_shift = (dash_offset % 1) * period
+
+            # closed shapes can handle overflow at the 0-point
+            if vmobject.is_closed():
+                dash_starts = [((i * period + phase_shift) % 1) for i in range(n)]
+                dash_ends = [
+                    ((i * period + dash_len + phase_shift) % 1) for i in range(n)
+                ]
+            else:
+                dash_starts = [
+                    ((i * period + phase_shift) % (1 + void_len)) for i in range(n)
+                ]
+                dash_ends = [
+                    ((i * period + dash_len + phase_shift) % (1 + void_len))
                     for i in range(n)
+                ]
+
+                # due to phase shift being [0...1] range, always the last dash element needs attention for overflow
+                # if an entire dash moves out of the shape end:
+                if dash_ends[-1] > 1 and dash_starts[-1] > 1:
+                    # remove the last element since it is out-of-bounds
+                    dash_ends.pop()
+                    dash_starts.pop()
+                else:
+                    # if it overflowed:
+                    if dash_ends[-1] < dash_len:
+                        # if the beginning of the piece is still in range
+                        if dash_starts[-1] < 1:
+                            dash_starts.append(0)
+                            dash_ends.append(dash_ends[-1])
+                            dash_ends[-2] = 1
+                        else:
+                            dash_starts[-1] = 0
+                    else:
+                        if dash_starts[-1] > (1 - dash_len):
+                            dash_ends[-1] = 1
```

Comment:
Again, matter of taste. Feel free to leave as-is.

```suggestion
                elif dash_ends[-1] < dash_len:  # if it overflowed
                    if dash_starts[-1] < 1:  # if the beginning of the piece is still in range
                        dash_starts.append(0)
                        dash_ends.append(dash_ends[-1])
                        dash_ends[-2] = 1
                    else:
                        dash_starts[-1] = 0
                elif dash_starts[-1] > (1 - dash_len):
                    dash_ends[-1] = 1
```